src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -122,25 +184,69 @@ def filter(
         -------
         KalmanPosterior
             Posterior distribution of the filtered output
+        info_dicts
+            list of dictionaries containing filtering information
+
+        See Also
+        --------
+        RegressionProblem: a regression problem data class
+        """
+
+        filtered_rvs = []
+        info_dicts = []
+
+        for rv, info in self.filter_generator(regression_problem, _previous_posterior):
+            filtered_rvs.append(rv)
+            info_dicts.append(info)
+
+        posterior = FilteringPosterior(
+            locations=regression_problem.locations,
+            states=filtered_rvs,
+            transition=self.dynamics_model,
+        )
+
+        return posterior, info_dicts
+
+    def filter_generator(
```

Comment:
naming question: is this a `filter_generator` or something along the lines of a `filter_solution_generator`?

Reply:
decision: `filtered_states_generator`